metrics-operator/controllers/common/providers/prometheus/prometheus.go

```diff
@@ -23,34 +23,116 @@ func (r *KeptnPrometheusProvider) EvaluateQuery(ctx context.Context, metric metr
 	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
 	defer cancel()
 
-	queryTime := time.Now().UTC()
-	r.Log.Info("Running query: /api/v1/query?query=" + metric.Spec.Query + "&time=" + queryTime.String())
 	client, err := promapi.NewClient(promapi.Config{Address: provider.Spec.TargetServer, Client: &r.HttpClient})
 	if err != nil {
 		return "", nil, err
 	}
+
 	api := prometheus.NewAPI(client)
-	result, w, err := api.Query(
+	if metric.Spec.Range != nil {
+		result, warnings, err := evaluateQueryWithRange(ctx, metric, r, api)
+		if err != nil {
+			return "", nil, err
+		}
+		if len(warnings) != 0 {
+			r.Log.Info("Prometheus API returned warnings: " + warnings[0])
+		}
+		return getResultForRange(result, r)
```

Comment:
i suggest to name the `getResultForRange`/`getResultForNoRange` functions to `getResultForMatrix`/`getResultForVector` to better reflect what they are doing